src/hook.c

```diff
@@ -342,10 +342,23 @@ static MH_STATUS Freeze(PFROZEN_THREADS pThreads, UINT pos, UINT action)
         for (i = 0; i < pThreads->size; ++i)
         {
             HANDLE hThread = OpenThread(THREAD_ACCESS, FALSE, pThreads->pItems[i]);
-            if (hThread != NULL)
+            if (hThread == NULL)
             {
-                SuspendThread(hThread);
-                ProcessThreadIPs(hThread, pos, action);
+                // Mark thread as not suspended, so it's not resumed later on.
+                pThreads->pItems[i] = 0;
+            }
+            else
+            {
+                DWORD result = SuspendThread(hThread);
+                if (result == 0xFFFFFFFF)
+                {
+                    // Mark thread as not suspended, so it's not resumed later on.
+                    pThreads->pItems[i] = 0;
+                }
+                else
+                {
+                    ProcessThreadIPs(hThread, pos, action);
+                }
                 CloseHandle(hThread);
             }
```

Comment:
Looks good. How about avoiding the duplicate code? Something like:
```cpp
            HANDLE hThread = OpenThread(THREAD_ACCESS, FALSE, pThreads->pItems[i]);
            BOOL suspended = FALSE;
            if (hThread != NULL)
            {
                DWORD result = SuspendThread(hThread);
                if (result != 0xFFFFFFFF)
                {
                    suspended = TRUE;
                    ProcessThreadIPs(hThread, pos, action);
                }
                CloseHandle(hThread);
            }

            if (!suspended)
            {
                // Mark thread as not suspended, so it's not resumed later on.
                pThreads->pItems[i] = 0;
            }
```

Reply:
lol yeah I thought about that, but it would introduce a variable, so I wasn't sure if that would get accepted, I'll change it, it felt a bit dirty with the duplicated block indeed. Sec.

Reply:
Done [df15207](https://github.com/TsudaKageyu/minhook/pull/134/commits/df15207abd62dd7e1a3088a68164d820fbf47e9a)